### PR TITLE
GC some unused code and less sync_all_executors.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

File: searchcore/src/tests/proton/documentdb/document_subdbs/document_subdbs_test.cpp [hunk 1]
```diff
@@ -319,7 +319,7 @@ struct FixtureBase
     }
     template <typename FunctionType>
     void runInMaster(FunctionType func) {
-        proton::test::runInMaster(_writeService, func);
+        proton::test::runInMasterAndSync(_writeService, func);
```
Comment (toregge) [inline searchcore/src/tests/proton/documentdb/document_subdbs/document_subdbs_test.cpp:322]: Consider renaming member function at line 321.

File: searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp [hunk 2]
```diff
@@ -513,7 +513,7 @@ struct FixtureBase
 
     template <typename FunctionType>
     void runInMaster(FunctionType func) {
-        test::runInMaster(_writeService, func);
+        test::runInMasterAndSync(_writeService, func);
```
Comment (toregge) [inline searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp:516]: Consider renaming member funciton name at line 515.

File: searchcore/src/tests/proton/documentdb/storeonlyfeedview/storeonlyfeedview_test.cpp [hunk 3]
```diff
@@ -242,7 +242,7 @@ struct FixtureBase {
 
     template <typename FunctionType>
     void runInMaster(FunctionType func) {
-        test::runInMaster(writeService, func);
+        test::runInMasterAndSync(writeService, func);
```
Comment (toregge) [inline searchcore/src/tests/proton/documentdb/storeonlyfeedview/storeonlyfeedview_test.cpp:245]: Consider renaming member function name at line 244.

File: searchcore/src/tests/proton/documentmetastore/lidreusedelayer/lidreusedelayer_test.cpp [hunk 4]
```diff
@@ -137,7 +137,7 @@ class Fixture
 
     template <typename FunctionType>
     void runInMaster(FunctionType func) {
-        test::runInMaster(_writeService, func);
+        test::runInMasterAndSync(_writeService, func);
```
Comment (toregge) [inline searchcore/src/tests/proton/documentmetastore/lidreusedelayer/lidreusedelayer_test.cpp:140]: Consider renaming member function name at line 139.

**Review by toregge — APPROVED** [review]

File: searchcore/src/tests/proton/documentdb/document_subdbs/document_subdbs_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/documentdb/document_subdbs/document_subdbs_test.cpp:322]: Done

File: searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp:516]: Done

File: searchcore/src/tests/proton/documentdb/storeonlyfeedview/storeonlyfeedview_test.cpp (hunk 3 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/documentdb/storeonlyfeedview/storeonlyfeedview_test.cpp:245]: Done

File: searchcore/src/tests/proton/documentmetastore/lidreusedelayer/lidreusedelayer_test.cpp (hunk 4 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/documentmetastore/lidreusedelayer/lidreusedelayer_test.cpp:140]: Done

**Pull request merged**